Need to add json tag in VolatileTime struct

<!--
Pro-tip: You can leave this block commented, and it still works!

Select the appropriate areas for your issue:

/area API


Classify what kind of issue this is:
/kind bug

-->

## Expected Behavior
Our controller is based on operator-sdk, and we use duck condition in our controller. We expect to upgrade to operator-sdk 0.17.1 without any problem.

## Actual Behavior
But it failed at `operator-sdk generate crds` step
```
operator-sdk generate crds
INFO[0000] Running CRD generator.
/usr/local/go/src/time/time.go:139:2: encountered struct field "wall" without JSON tag in type "Time"
/usr/local/go/src/time/time.go:140:2: encountered struct field "ext" without JSON tag in type "Time"
/usr/local/go/src/time/time.go:147:2: encountered struct field "loc" without JSON tag in type "Time"
/Users/mattcui/myspace/tenant-domain-controller/pkg/mod/knative.dev/pkg@v0.0.0-20191107185656-884d50f09454/apis/volatile_time.go:27:2: encountered struct field "Inner" without JSON tag in type "VolatileTime"
FATA[0001] error generating CRDs from APIs in pkg/apis: error generating CRD manifests: error running CRD generator: not all generators ran successfully
```

## How to fix
I think an easy way to fix it is to add json tag for VolatileTime struct at https://github.com/knative/pkg/blob/master/apis/volatile_time.go#L32, it should be changed to 

```
type VolatileTime struct {
	Inner metav1.Time `json:"inner"`
}
```

I could submit a PR for it. Thanks.

We also have similar requirement where we want to leverage the `Status`. We have a controller generated using kubebuilder 2.3 

```
import (
	corev1 "k8s.io/api/core/v1"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
	duckv1 "knative.dev/pkg/apis/duck/v1"
)

type DeploymentPayloadStatus struct {

	duckv1.Status `json:",inline"`
}
```
Currently the code generator fails 

/home/raushank/go/bin/controller-gen "crd:trivialVersions=true" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
/home/raushank/go/pkg/mod/knative.dev/pkg@v0.0.0-20200730180200-bd4b7612364d/apis/volatile_time.go:32:2: encountered struct field "Inner" without JSON tag in type "VolatileTime"
Error: not all generators ran successfully

--- a/apis/volatile_time.go
+++ b/apis/volatile_time.go
@@ -29,7 +29,7 @@ import (
 // Note, go-cmp will still return inequality, see unit test if you
 // need this behavior for go-cmp.
 type VolatileTime struct {
-	Inner metav1.Time
+	Inner metav1.Time `json:",inline"`
 }
 
 // MarshalJSON implements the json.Marshaler interface.

